notebooks/216-machine-translation/machine-translation-demo.ipynb

```diff
@@ -0,0 +1,304 @@
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "! omz_downloader --name  machine-translation-nar-en-de-0002"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {
+    "pycharm": {
+     "is_executing": true,
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "import time\n",
+    "from openvino.runtime import Core\n",
+    "import numpy as np\n",
+    "import itertools\n",
+    "from tokenizers import SentencePieceBPETokenizer"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "## Load and configure the model\n",
+    "The model is now available in the `intel/` folder. Below we load and configure its inputs and outputs."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "core = Core()\n",
+    "model = core.read_model('intel/machine-translation-nar-en-de-0002/FP32/machine-translation-nar-en-de-0002.xml')\n",
+    "compiled_model = core.compile_model(model)\n",
+    "infer_request = compiled_model.create_infer_request()\n",
+    "input_name = \"tokens\"\n",
+    "output_name = \"pred\"\n",
+    "model.output(output_name)\n",
+    "max_tokens = model.input(input_name).shape[1]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "## Load tokenizers\n",
+    "Before providing the model with an input sentence, the input needs to be transformed into tokens the model understands. We must also transform the output into a sentence that we can read.\n",
+    "\n",
+    "Initialize the tokenizer for the input `src_tokenizer` and the output `tgt_tokenizer`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "src_tokenizer = SentencePieceBPETokenizer.from_file(\n",
+    "    'intel/machine-translation-nar-en-de-0002/tokenizer_src/vocab.json',\n",
+    "    'intel/machine-translation-nar-en-de-0002/tokenizer_src/merges.txt'\n",
+    ")\n",
+    "tgt_tokenizer = SentencePieceBPETokenizer.from_file(\n",
+    "    'intel/machine-translation-nar-en-de-0002/tokenizer_tgt/vocab.json',\n",
+    "    'intel/machine-translation-nar-en-de-0002/tokenizer_tgt/merges.txt'\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "## Perform translation\n",
+    "The following function translates a sentence in English to German."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def translate(sentence):\n",
```

Comment:
Please use docstring format and type hints, for all classes, functions, methods. An example could be found here: https://user-images.githubusercontent.com/10940214/163566002-9e552d71-ba99-4d88-afb8-8c439b90aaac.png